swap/swap.go

```diff
@@ -346,12 +346,14 @@ func (s *Swap) loadLastSentCheque(peer enode.ID) (err error) {
 	var cheque *Cheque
 	if _, ok := s.cheques[peer]; !ok {
 		err = s.store.Get(sentChequeKey(peer), &cheque)
-		s.cheques[peer] = cheque
+		if err != nil && err != state.ErrNotFound {
+			s.cheques[peer] = cheque
```

Comment:
@holisticode we are assigning the cheque when `err` is not `nil`. shouldn't it be the other way around?

Reply:
You are right, great catch, the problem with the previous code was that there was no check, thus we would get `nil` cheques for peers if there was not a last cheque yet for a peer:

`s.cheques[78asdf.....] = nil`

But the fix I sent is wrong. Thanks. Will send an update.